enhancements/20230807-rancher-vcluster-addon.md

```diff
@@ -0,0 +1,80 @@
+* fine grained rbac
+* federation with external identity providers
+* provisioning/management of downstream k8s clusters running on harvester
+
+This requires that a user deploy an external Rancher cluster on a single VM or multiple VM's for HA setup. A lot of users not already having a rancher install end up running these VM's on Harvester to host the Rancher installation. 
+
+This approach as the following limitations:
+* initial work requirement for provisioning of infra and managaging the infra on an ongoing basis
+* even for a single node installations, a user generally ends up reserving 2vCPU and 8GB memory. This may be not be feasible for low compute / development environments.
+
+### API changes
+No API changes needed. The rancher-vcluster packaging will be made available as an external addon.
+
+## Design
+The `rancher-vcluster` addon leverages [vcluster](https://www.vcluster.com), which as per official docs serves the following purpose:
+```
+Virtual clusters are fully working Kubernetes clusters that run on top of other Kubernetes clusters. Compared to fully separate "real" clusters, virtual clusters reuse worker nodes and networking of the host cluster. They have their own control plane and schedule all workloads into a single namespace of the host cluster. Like virtual machines, virtual clusters partition a single physical cluster into multiple separate ones.
+```
+
+### Implementation Overview
+The `rancher-vcluster` addon is hosted in a repository for [experimental-addons](https://github.com/harvester/experimental-addons).
+
+To get started the end user needs to apply the manifest manually to their harvester cluster:
+
+```
+kubectl apply -f https://raw.githubusercontent.com/harvester/experimental-addons/main/rancher-vcluster/rancher-vcluster.yaml
+```
+
+This can be enable and configured from harvester addons page:
+
+![](./20230807-rancher-vcluster-addon/rancher-vcluster-config.png)
+
+A user needs to provide a valid `hostname` and `bootstrap password` to get started.
+
+**NOTE** vcluster is deployed with ingress to underlying host cluster enabled. As a result of this `hostname` needs to a valid DNS record pointing to the harvester VIP.
+
+When enabled a helm chart for vcluster is deployed in the background. 
+
+We leverage the advanced [init-manifests](https://www.vcluster.com/docs/operator/init-manifests) capability of vcluster to inject the following components into vcluster:
+* cert-manager
+* rancher
+
+At present the installed `rancher` leverages a self signed certificate, and this can be extended in the future to allow users to brign their own certificates.
+
+Once the addon is deployed, a rancher endpoint is available at the user defined `hostname`.
+
+### Test plan
+
+* Enable `rancher-vcluster` addon.
+* In a few minutes a rancher installation should be accessible at the predefined `hostname`
+* Import underlying harvester into `rancher-vcluster` and perform MCM tests using this rancher.
+
```

Comment:
Please add `Disable rancher-vcluster`  and `Re-enable rancher v-cluster`
When `Disabled`, the Rancher related data is stored in in PV, will not be lost

When `Enable` again, the Rancher will recover from previous settings and saved data. 

At this case, does `Bootstrap Password` field still make sense? and can `Hostname` and `Rancher version` be changed when re-enabling?

In certain cases, the `rancher-vcluster` may be corrupted or user wants to throw it away, let's point a way to hardly reset the `rancher-vcluster`, namely revmove the PV manually.

Reply:
And, unlike other addons, it seems the `rancher-vcluster` may support multi-instances in the future, each `Hostname` will be the identification of the Rancher instance.